### PR TITLE
[Hub Generated] Review request for Route to add version preview/2023-10-01-preview

### PR DESCRIPTION
**Pull request by koyasu221b**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://openapihub.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/koyasu221b-maps-Route-2023-10-01-preview...main/).
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:  [link](https://microsoftapc.sharepoint.com/:p:/t/AzureMapsTW/EQCW7Kd_oW9Is6Egn56FUXkBtKQ969rSEqc2UkIBcJWAow)
* Previous API Spec Doc: https://github.com/Azure/azure-rest-api-specs/pull/23044
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

File: specification/maps/data-plane/Route/preview/2023-10-01-preview/route.json [hunk 1]
```diff
@@ -278,7 +278,7 @@
               {
                 "value": "fastestAvoidClosureWithoutTraffic",
                 "name": "FastestAvoidClosureWithoutTraffic",
-                "description": "The route is calculated to minimize the time and avoid road closures. No traffic information except for road closures is used in the calculation."
+                "description": "The route is calculated to minimize the time and avoid road closures. No traffic information except for road closures is used in the calculation. `Note`: Only supported for driving and walking travelMode."
```
Comment (FarazGIS) [inline specification/maps/data-plane/Route/preview/2023-10-01-preview/route.json:281]: @koyasu221b Can you please check if Bing routing gives a response for walking for fastestAvoidClosureWithoutTraffic

File: specification/maps/data-plane/Route/preview/2023-10-01-preview/route.json [hunk 2]
```diff
@@ -312,8 +312,8 @@
               "modelAsString": true,
               "values": [
                 {
-                  "value": "highways",
-                  "description": "Avoids the use of highways in the route. `Note`: Only supported for driving travelMode."
+                  "value": "limitedAccessHighway",
+                  "description": "Avoids the use of limited access highway in the route. `Note`: Only supported for driving and truck travelMode."
                 },
```
Comment (FarazGIS) [inline specification/maps/data-plane/Route/preview/2023-10-01-preview/route.json:317]: @koyasu221b We don't need to mention only supported for driving and truck if it's supported by both travel modes like ferries and toll road and it is implied that avoid highways would not be applicable to walking.

File: specification/maps/data-plane/Route/preview/2023-10-01-preview/route.json [hunk 3]
```diff
@@ -299,7 +299,7 @@
           "items": {
             "type": "string",
             "enum": [
-              "highways",
+              "limitedAccessHighway",
```
Comment (winest) [inline specification/maps/data-plane/Route/preview/2023-10-01-preview/route.json:302]: limitedAccessHighway"s" for consistency with others

**Review by winest — CHANGES_REQUESTED** [review]

File: specification/maps/data-plane/Route/preview/2023-10-01-preview/route.json [hunk 4]
```diff
@@ -299,7 +299,7 @@
           "items": {
```
[the remark this answers is not in the document]
Comment (koyasu221b) [reply specification/maps/data-plane/Route/preview/2023-10-01-preview/route.json:299]: fixed

**Review by winest — CHANGES_REQUESTED** [review]
Line 423 also need to include ErrorDetails for DirectionsResponse

**Comment by koyasu221b** [comment]
> Line 423 also need to include ErrorDetails for DirectionsResponse

After our discussion, we think it's not required

**Review by winest — APPROVED** [review]

**Review by JeffreyRichter — APPROVED** [review]

**Pull request merged**